core/src/main/kotlin/behavior/channel/StageChannelBehavior.kt

```diff
@@ -0,0 +1,76 @@
+package dev.kord.core.behavior.channel
+
+import dev.kord.common.entity.Snowflake
+import dev.kord.core.Kord
+import dev.kord.core.cache.data.ChannelData
+import dev.kord.core.entity.channel.Channel
+import dev.kord.core.entity.channel.StageChannel
+import dev.kord.core.supplier.EntitySupplier
+import dev.kord.core.supplier.EntitySupplyStrategy
+import dev.kord.rest.builder.channel.StageVoiceChannelModifyBuilder
+import dev.kord.rest.builder.guild.CurrentVoiceStateModifyBuilder
+import dev.kord.rest.builder.guild.VoiceStateModifyBuilder
+import dev.kord.rest.service.modifyCurrentVoiceState
+import dev.kord.rest.service.modifyVoiceState
+import dev.kord.rest.service.patchStageVoiceChannel
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+
+interface StageChannelBehavior : BaseVoiceChannelBehavior {
+
+    override fun withStrategy(
+        strategy: EntitySupplyStrategy<*>
+    ): StageChannelBehavior {
+        return StageChannelBehavior(id, guildId, kord, strategy.supply(kord))
+    }
+
+    fun StageChannelBehavior(
+        id: Snowflake,
+        guildId: Snowflake,
+        kord: Kord,
+        supplier: EntitySupplier = kord.defaultSupplier
+    ): StageChannelBehavior = object : StageChannelBehavior {
+        override val guildId: Snowflake get() = guildId
+        override val id: Snowflake get() = id
+        override val kord get() = kord
+        override val supplier get() = supplier
+    }
```

Comment:
This is still here, while it should be defined outside the interface.
```suggestion
```

Reply:
I'm sorry didn't notice that I've copied the declaration even though you have mentioned it earlier